src/accountsdb/download.zig

```diff
@@ -302,81 +304,93 @@ pub fn downloadSnapshotsFromGossip(
 }
 
 const DownloadProgress = struct {
-    mmap: []align(std.mem.page_size) u8,
-    download_size: usize,
+    file: std.fs.File,
     min_mb_per_second: ?usize,
     logger: Logger,
 
     mb_timer: std.time.Timer,
-    bytes_read: usize = 0,
-    file_memory_index: usize = 0,
+    bytes_read: u64 = 0,
+    total_read: u64 = 0,
     has_checked_speed: bool = false,
 
     const Self = @This();
 
-    pub fn init(
+    fn init(
         logger: Logger,
         output_dir: std.fs.Dir,
         filename: []const u8,
         download_size: usize,
         min_mb_per_second: ?usize,
     ) !Self {
-        const file = try output_dir.createFile(filename, .{ .read = true });
-        defer file.close();
+        const file = try output_dir.createFile(filename, .{});
 
         // resize the file
         try file.seekTo(download_size - 1);
         _ = try file.write(&[_]u8{1});
         try file.seekTo(0);
 
-        const file_memory = try std.posix.mmap(
-            null,
-            download_size,
-            std.posix.PROT.READ | std.posix.PROT.WRITE,
-            std.posix.MAP{ .TYPE = .SHARED },
-            file.handle,
-            0,
-        );
-
         return .{
             .logger = logger,
-            .mmap = file_memory,
-            .download_size = download_size,
+            .file = file,
             .min_mb_per_second = min_mb_per_second,
-            .mb_timer = try std.time.Timer.start(),
+            .mb_timer = undefined,
         };
```

Comment:
changed from undefined to use a helper function -- it will always be used so optional doesnt make sense -- undefined doesnt feel right -- 
[74d9015](https://github.com/Syndica/sig/pull/316/commits/74d9015aec1985cf2d2d824107ae422aa98b7af5)